tfjs-core/src/ops/one_hot.ts

```diff
@@ -56,17 +56,16 @@ function oneHot_(
 
   const forward: ForwardFunc<Tensor> = (backend, save) => {
     save([$indices]);
-    return reshape(
-        backend.oneHot($indices as Tensor1D, depth, onValue, offValue),
-        outShape);
+    return backend.oneHot($indices as Tensor1D, depth, onValue, offValue);
   };
 
   const inputs: OneHotInputs = {indices: $indices};
   const attrs: OneHotAttrs = {depth, onValue, offValue};
 
-  return ENGINE.runKernelFunc(
+  const res = ENGINE.runKernelFunc(
       forward, inputs as unknown as NamedTensorMap, null /* grad */, OneHot,
       attrs as unknown as NamedAttrMap);
+  return reshape(res, outShape);
```

Comment:
We would like things like reshape to happen within the kernel as well. Could you add logic to reshape in the WASM OneHot typescript implementation?

Reply:
I modified the one hot op in core a bit so that flatten is moved into the forward function. To get the correct output shape in the WASM op, I need to pass in the original indices tensor (i.e. not flattened).  I hope this is fine.